keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/cluster/TitusClusterHandler.kt

```diff
@@ -0,0 +1,169 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.api.titus.cluster
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceId
+import com.netflix.spinnaker.keel.api.ResourceKind
+import com.netflix.spinnaker.keel.api.serviceAccount
+import com.netflix.spinnaker.keel.api.titus.CLOUD_PROVIDER
+import com.netflix.spinnaker.keel.api.titus.SPINNAKER_TITUS_API_V1
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.clouddriver.model.TitusActiveServerGroup
+import com.netflix.spinnaker.keel.diff.ResourceDiff
+import com.netflix.spinnaker.keel.events.Task
+import com.netflix.spinnaker.keel.model.ClusterDependencies
+import com.netflix.spinnaker.keel.orca.OrcaService
+import com.netflix.spinnaker.keel.plugin.Resolver
+import com.netflix.spinnaker.keel.plugin.ResourceHandler
+import com.netflix.spinnaker.keel.retrofit.isNotFound
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import org.springframework.context.ApplicationEventPublisher
+import retrofit2.HttpException
+import java.time.Clock
+
+class TitusClusterHandler(
+  private val cloudDriverService: CloudDriverService,
+  private val cloudDriverCache: CloudDriverCache,
+  private val orcaService: OrcaService,
+  private val clock: Clock,
+  private val publisher: ApplicationEventPublisher,
+  objectMapper: ObjectMapper,
+  resolvers: List<Resolver<*>>
+) : ResourceHandler<TitusClusterSpec, Map<String, TitusServerGroup>>(objectMapper, resolvers) {
+
+  override val apiVersion = SPINNAKER_TITUS_API_V1
+  override val supportedKind = ResourceKind(
+    group = apiVersion.group,
+    singular = "titus-cluster",
+    plural = "titus-clusters"
+  ) to TitusClusterSpec::class.java
+
+  override suspend fun toResolvedType(resource: Resource<TitusClusterSpec>): Map<String, TitusServerGroup> =
+    with(resource.spec) {
+      resolve().byRegion()
+    }
+
+  override suspend fun current(resource: Resource<TitusClusterSpec>): Map<String, TitusServerGroup>? =
+    cloudDriverService
+      .getServerGroups(resource)
+      .byRegion()
+
+  override suspend fun create(resource: Resource<TitusClusterSpec>, resourceDiff: ResourceDiff<Map<String, TitusServerGroup>>): List<Task> {
+    return super.create(resource, resourceDiff)
+  }
+
+  override suspend fun update(resource: Resource<TitusClusterSpec>, resourceDiff: ResourceDiff<Map<String, TitusServerGroup>>): List<Task> {
+    return super.update(resource, resourceDiff)
+  }
+
+  override suspend fun upsert(resource: Resource<TitusClusterSpec>, resourceDiff: ResourceDiff<Map<String, TitusServerGroup>>): List<Task> {
+    return super.upsert(resource, resourceDiff)
+  }
```

Comment:
You shouldn't need to override all 3 of these. See comments on interface.

Reply:
Oh that's just what intellij generated for me. You're right.